addons/web/static/src/core/utils/hooks.js

```diff
@@ -164,6 +160,43 @@ export function useService(serviceName) {
     return service;
 }
 
+/**
+ * Import a service into a component
+ *
+ * @template {keyof import("services").ServiceFactories} K
+ * @param {K} serviceName
+ * @param {Function} [onDelayedServiceReady] - To be called when a delayed stateful service is available
+ * @returns {import("services").ServiceFactories[K]}
+ */
+export function useService(serviceName, onDelayedServiceReady) {
+    const component = useComponent();
+    const { services } = component.env;
+    if (serviceName in services) {
+        return decorateService(serviceName, services[serviceName], component);
+    }
+    const tempState = useState({});
+    onWillStart(async () => {
+        await new Promise((resolve, reject) => {
+            setTimeout(() => {
+                if (!(serviceName in services)) {
+                    reject(
+                        new Error(
+                            `Service ${serviceName} is not available in "${component.constructor.name}".`
+                        )
+                    );
+                } else {
+                    resolve();
+                }
+            }, 0);
+        });
+        if (typeof onDelayedServiceReady === "function") {
+            onDelayedServiceReady(Object.setPrototypeOf(tempState, services[serviceName]));
```

Comment:
Should not `setPrototype` there: the prototype is not observed in OWL.reactive.

Instead should:
```js
tempState.service = services[serviceName];
onDelayedServiceReady(tempState.service);
```